### PR TITLE
Refactor View Code

### PR DESCRIPTION
**Pull request by AvlWx2014**
Begin refactoring view code from the E2080 project, starting with timing related javascript functions. There were 3 view files with the same javascript functions declared in each. Extract those to `review_time.js` and use `<%= javascript_include_tag "review_time" %>` to include those functions in the relevant pages.

### CODE REVIEW

**Comment by AvlWx2014** [comment]
Can you remember what function you were talking about? It doesn't show up here.

File: app/views/submitted_content/_submitted_files.html.erb [hunk 1]
```diff
@@ -1,120 +1,149 @@
...
+                if ($("#restart_file_timing_flag").val() === 'true') {
+                    filesToUpdate.forEach(function (file) {
+                        if (filesToUpdate[file]) {
+                            startTime(map_id, round, file);
+                        }
+                    });
+
+                    $("#restart_file_timing_flag").val('false');
+                }
             }
```
Comment (boergerj) [inline app/views/submitted_content/_submitted_files.html.erb]: Earlier comment refers to this function being moved to the bottom with checkWindowStatus

File: app/views/submitted_content/_hyperlink.html.erb [hunk 2]
```diff
@@ -1,146 +1,124 @@
...
+                        linksToUpdate[address] = false; // sets link in tracking array to false
+                        clearInterval(link_timer); // removes check for window open/close
+                    }
+                    timeoutAction(timeout_flag); // if timeout flag is set, times out page
+                }
+            });
+
+            function markEndAndCloseWindows() {
+                markEndTime(map_id, round); // saves end time of all open links
+                windowManager.closeAll()
```
Comment (boergerj) [inline app/views/submitted_content/_hyperlink.html.erb]: I would move this function to the bottom of this call where timeoutAction is

File: app/views/submitted_content/_hyperlink.html.erb [hunk 3]
```diff
@@ -1,146 +1,124 @@
...
+
+                // Checks if window closed to then track end time
+                function checkWindowStatus() {
+                    let timeout_flag = $('#timeout_flag').val() == 'true'; // checks for timeout flag == true and if so sets
+                    if (window_handler.closed) { // if window closed
+                        endTime(map_id, round, address); // save end time of link
+                        linksToUpdate[address] = false; // sets link in tracking array to false
+                        clearInterval(link_timer); // removes check for window open/close
+                    }
+                    timeoutAction(timeout_flag); // if timeout flag is set, times out page
```
Comment (boergerj) [inline app/views/submitted_content/_hyperlink.html.erb:111]: Does this need to be inside the .click() call?

File: app/views/submitted_content/_hyperlink.html.erb (hunk 2 above)
Comment (AvlWx2014) [reply app/views/submitted_content/_hyperlink.html.erb]: I was wondering about this, I didn't know what good practice is, but I think [this](https://eslint.org/docs/rules/no-use-before-define) is a valid opinion to follow. So I moved this and timeoutAction to the top of the block, after the variable declarations.

File: app/views/submitted_content/_hyperlink.html.erb (hunk 3 above)
Comment (AvlWx2014) [reply app/views/submitted_content/_hyperlink.html.erb:111]: Sadly, yes. It references the `window_handler` variable which is scoped to the that function.

**Pull request merged**